giftcompare/gifts/views.py

```diff
@@ -21,11 +21,20 @@ def post(self, request):
 class GiftDetail(APIView):
     def get_object(self, pk):
         return Gift.objects.get(pk=pk)
-    
+
     def get(self, request, pk):
         try:
             gift = self.get_object(pk)
             serializer = GiftSerializer(gift)
             return Response(serializer.data)
         except Gift.DoesNotExist:
-            return Response(status=status.HTTP_404_NOT_FOUND)
```

Comment:
Just a note - not need to implement - you can resolve this comment without code changes.
I would, and I have done it in my crowdfunding project, have raised the 404 error in the general get_object function (after the class)
If you raise it there, you cover both the GET and the PUT methods.

This is what I have done (for the ProjectDetail and PledgeDetail):
from django.http import Http404

def get_object(self, pk):
        try:
            project = Project.objects.get(pk=pk)
            self.check_object_permissions(self.request, project)
            return project
        except Project.DoesNotExist:
            raise Http404
            
 then I implemented my GET and POST functions.
 just wanted to share code reusability insight.

Reply:
Discussed with Alisha and happy to resolve.